ebuild-writing/eapi/text.xml

```diff
@@ -62,8 +62,34 @@ EAPI-conditional code)
 When writing new ebuilds developers can choose whatever EAPI they think
 is the best.  Using the features of the latest EAPI is encouraged.
 </p>
+</body>
+
+<subsection>
+<title>Upgrade path</title>
+<body>
+
+<p>
+Gentoo policy is to support upgrades for installations at least a year old
+with no/little intervention and up to 2 years old with minor intervention. To
+achieve this, developers must avoid using the latest EAPI in ebuilds within
+the <c>@system</c> set (see <uri link="::general-concepts/dependencies/#implicit-system-dependency"/>)
+</p>
+
+<p>
+The Base System project has
+<uri link="https://wiki.gentoo.org/wiki/Project:Base#Rules_and_limitations">rules</uri>
+governing their use of newer EAPIs, as does the
+<uri link="https://dev.gentoo.org/~mgorny/python-guide/package-maintenance.html#porting-packages-to-a-new-eapi">Python project</uri>.
+</p>
+
+<p>
+It is also convention that blockers within ebuilds are retained for at least
+2 years after the replacement / newer version is marked stable to avoid file
+collisions for users upgrading older systems.
```

Comment:
Not sure if that's what we're actually doing. Blocks stay in place as long as the ebuild that is blocked against is still in the tree, and some time after its removal.

Presumably the two years should count after removal of the last ebuild matching the block.

Reply:
Well, pkgcheck goes from 2 years after the last match is removed, yeah. I can change it to that.